src/rdo.rs

```diff
@@ -328,20 +330,27 @@ pub fn rdo_mode_decision(
           false
         );
 
-        if rd < best_rd {
-          best_rd = rd;
-          best_mode_luma = luma_mode;
-          best_mode_chroma = chroma_mode;
-          best_cfl_params = cfl;
-          best_ref_frame = ref_frame;
-          best_mv = mv;
-          best_skip = skip;
-        }
-
         cw.rollback(&cw_checkpoint);
-      }
-    }
-  }
+
+        (rd, luma_mode, chroma_mode, cfl, ref_frame, mv, skip)
+      }).fold((best_rd, best_mode_luma, best_mode_chroma, best_cfl_params, best_ref_frame, best_mv, best_skip),
```

Comment:
I read what 'fold' is for from https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.fold,
there it says fold accumulates the collection or sequence (thru the iteration) of values to the initial value.
Is initial value here a tuple? i.e. (best_rd, best_mode_luma, best_mode_chroma, best_cfl_params, best_ref_frame, best_mv, best_skip), where their values given at lines from 235 above?

Reply:
yes :)